File: src/modules/presentation/io/tripled/social/client/presentation/controller/PostController.java
package io.tripled.social.client.presentation.controller;

import io.tripled.social.client.application.PostMessageUseCase;
import io.tripled.social.client.presentation.Controller;
import io.tripled.social.client.presentation.Output;
import io.tripled.social.client.presentation.request.PostCliRequest;

public class PostController implements Controller<PostCliRequest> {

  private PostMessageUseCase postMessageUseCase;

  public PostController(PostMessageUseCase postMessageUseCase) {
    this.postMessageUseCase = postMessageUseCase;
  }

  @Override
  public void execute(PostCliRequest request, Output output) {
    output.print(postMessageUseCase.postMessage(request.getUserName(), request.getMessage()));
  }
}

File: src/main/kotlin/io/tripled/social/client/infrastructure/cli/controller/PostController.kt
package io.tripled.social.client.infrastructure.cli.controller

import io.tripled.social.client.application.PostMessage
import io.tripled.social.client.infrastructure.cli.Controller
import io.tripled.social.client.infrastructure.cli.Output
import io.tripled.social.client.infrastructure.cli.request.PostCliRequest

class PostController(private val postMessage: PostMessage) : Controller<PostCliRequest> {
    override fun execute(request: PostCliRequest, output: Output) {
        output.print(postMessage.postMessage(request.userName, request.message))
    }
}
